examples/new_actions.py

```diff
@@ -12,169 +12,132 @@
 We will use the strafe action outline in the habitat_sim example
 """
 
-import attr
+from dataclasses import dataclass
+
+import magnum as mn
 import numpy as np
 
 import habitat
-import habitat_sim
 from habitat.config.default_structured_configs import ActionConfig
-from habitat.sims.habitat_simulator.actions import (
-    HabitatSimActions,
-    HabitatSimV1ActionSpaceConfiguration,
-)
 from habitat.tasks.nav.nav import SimulatorTaskAction
 
 
-@attr.s(auto_attribs=True, slots=True)
-class NoisyStrafeActuationSpec:
-    move_amount: float
-    # Classic strafing is to move perpendicular (90 deg) to the forward direction
-    strafe_angle: float = 90.0
-    noise_amount: float = 0.05
+# This is the configuration for our action.
+@dataclass
+class StrafeActionConfig(ActionConfig):
+    move_amount: float = 0.0  # We will change this in the configuration
+    noise_amount: float = 0.0
 
 
-def _strafe_impl(
-    scene_node: habitat_sim.SceneNode,
+# This is a helper that implements strafing that we will use in our actions
+def _strafe_body(
+    sim,
     move_amount: float,
-    strafe_angle: float,
+    strafe_angle_degrees: float,
     noise_amount: float,
 ):
-    forward_ax = (
-        np.array(scene_node.absolute_transformation().rotation_scaling())
-        @ habitat_sim.geo.FRONT
+    # Get the state of the agent
+    agent_state = sim.get_agent_state()
+    # Convert from np.quaternion (quaternion.quaternion) to mn.Quaternion
+    normalized_quaternion = agent_state.rotation
+    agent_mn_quat = mn.Quaternion(
+        normalized_quaternion.imag, normalized_quaternion.real
     )
-    strafe_angle = np.deg2rad(strafe_angle)
+    forward = agent_mn_quat.transform_vector(-mn.Vector3.z_axis())
+    strafe_angle = np.deg2rad(strafe_angle_degrees)
```

Comment:
Nit: I think you can do this entire with Magnum calls.

Reply:
Besides random uniform and deg2rad, everything is magnum calls.
These were there in the original implementation, so I think I will leave them here.